esi_leap/objects/offer.py

```diff
@@ -58,6 +62,44 @@ def get_all_by_status(cls, context, status):
             context, status)
         return cls._from_db_object_list(context, db_offers)
 
+    @classmethod
+    def send(cls, context, offer_uuid):
+        db_offer = cls.dbapi.offer_get(context, offer_uuid)
+        o = cls._from_db_object(context, cls(), db_offer)
+
+        auth = v3.Password(auth_url='http://localhost:5000/v3',
+                           username='flocx-provider',
+                           password='password',
+                           project_name='service',
+                           user_domain_id='default',
+                           project_domain_id='default')
+        sess = session.Session(auth=auth)
+
+        offer_dict = o.to_dict()
+        # modifiy the offer data in order to be compatible with flocx-market POST offer api
+        # change fields name
+        offer_dict['start_time'] = offer_dict.pop('start_date')
+        offer_dict['end_time'] = offer_dict.pop('end_date')
+        offer_dict['server_config'] = offer_dict.pop('properties')
+        offer_dict['server_id'] = offer_dict.pop('resource_uuid')
+        offer_dict['provider_id'] = offer_dict.pop('uuid')
+        # remove unnecessary feilds
+        offer_dict.pop('created_at')
+        offer_dict.pop('updated_at')
+        offer_dict.pop('id')
+        offer_dict.pop('project_id')
+        offer_dict.pop('resource_type')
+        # fake fields
+        offer_dict['creator_id'] = 'Alice1992'
+        offer_dict['marketplace_date_created'] = '2016-07-16T19:20:30'
+        offer_dict['cost'] = 11.5
```

Comment:
I'd advise creating another method for Offer called something like `to_marketplace_dict` which creates the dict that you're going to pass below. Then this code (along with the other changes) becomes something like:

```
o = Offer.get()
marketplace_offer_dict = o.to_marketplace_dict()
<create client, send the dict>
```

Reply:
I'll do that!